frontend/src/components/ArticlePreview.js

```diff
@@ -1,16 +1,29 @@
 import { Link } from "react-router-dom";
 import { useArticlesContext } from "../hooks/useArticlesContext";
+import { useAuthContext } from "../hooks/useAuthContext";
 
 const ArticlePreview = ({ article }) => {
+  const { user } = useAuthContext();
   const { dispatch } = useArticlesContext();
+
   const handleClick = async () => {
-    const response = await fetch("/api/articles/" + article._id, {
-      method: "DELETE",
-    });
-    const json = await response.json();
+    const deleteArticle = async () => {
+      const response = await fetch("/api/articles/" + article._id, {
+        method: "DELETE",
+        headers: {
+          Authorization: `Bearer ${user.token}`,
+        },
+      });
+      const json = await response.json();
+
+      if (response.ok) {
+        dispatch({ type: "DELETE", payload: json });
+      }
```

Comment:
No else here?

Reply:
Mmm yes I will add something here, don't resolve this one

Reply:
[6066c2c](https://github.com/atxr/iscsc.fr/commit/6066c2c857cf4bf7fdd23da864a5dcc8690317df)